lib/formatters/html.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @fileoverview HTML reporter
+ * @author Julian Laval
+ * @copyright 2015 Julian Laval. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Given a word and a count, append an s if count is not one.
+ * @param {string} word A word in its singular form.
+ * @param {int} count A number controlling whether word should be pluralized.
+ * @returns {string} The original word with an s on the end if count is not one.
+ */
+function pluralize(word, count) {
+    return (count === 1 ? word : word + "s");
+}
+
+/**
+ * Escape text being outputted as HTML
+ * @param {string} text Text that needs to be escaped
+ * @returns {string} The escaped text
+ */
+function escapeHTML(text) {
+    var map = {
+        "&": "&amp;",
+        "<": "&lt;",
+        ">": "&gt;",
+        "\"": "&quot;",
+        "'": "&#039;"
+    };
+    return text.replace(/[&<>"']/g, function(m) {
+        return map[m];
+    });
+}
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results) {
+
+    var output = "<!DOCTYPE html><head><title>ESLint Report</title><style>"
```

Comment:
Would it be easier to use Mustache to generate the HTML? This seems like a headache. :)